evcouplings/align/protocol.py

```diff
@@ -1013,6 +1115,257 @@ def jackhmmer_search(**kwargs):
+    check_required(
+        kwargs,
+        [
+            "prefix", "sequence_id", "alignment_file",
+            "use_bitscores", "domain_threshold", "sequence_threshold",
+            "database", "iterations", "cpu", "nobias", "reuse_alignment",
+            "checkpoints_hmm", "checkpoints_ali", "hmmbuild",
+            "hmmsearch", "extract_annotation"
+        ]
+    )
+    prefix = kwargs["prefix"]
+
+    # make sure output directory exists
+    create_prefix_folders(prefix)
+
+    # prepare input alignment for hmmbuild
+    focus_fasta_file, target_sequence_file, region_start, region_end = \
+    _format_alignment_for_hmmbuild(
+        kwargs["alignment_file"], **kwargs
+    )
+
+    # run hmmbuild_and_search... allow to reuse pre-exisiting
+    # Stockholm alignment file here
+    ali_outcfg_file = prefix + ".align_hmmbuild_and_search.outcfg"
+
+    # determine if to rerun, only possible if previous results
+    # were stored in ali_outcfg_file
+    if kwargs["reuse_alignment"] and valid_file(ali_outcfg_file):
+        ali = read_config_file(ali_outcfg_file)
+
+        # check if the alignment file itself is also there
+        verify_resources(
+            "Tried to reuse alignment, but empty or "
+            "does not exist",
+            ali["alignment"], ali["domtblout"]
+        )
+    else:
+        # otherwise, we have to run the alignment
+        # modify search thresholds to be suitable for jackhmmer
+        sequence_length = region_end - region_start + 1 
+        seq_threshold, domain_threshold = search_thresholds(
+            kwargs["use_bitscores"],
+            kwargs["sequence_threshold"],
+            kwargs["domain_threshold"],
+            sequence_length
+        )
+
+        # create the hmm
+
+        hmmbuild_result = at.run_hmmbuild(
+            alignment_file=focus_fasta_file,
+            prefix=prefix,
+            symfrac=SYMFRAC_HMMBUILD,
+            cpu=kwargs["cpu"],
+            binary=kwargs["hmmbuild"],
+        )
+        hmmfile = hmmbuild_result.hmmfile
+
+        # run the alignment from the hmm
+        ali = at.run_hmmsearch(
+            hmmfile=hmmfile,
+            database=kwargs[kwargs["database"]],
+            prefix=prefix,
+            use_bitscores=kwargs["use_bitscores"],
+            domain_threshold=domain_threshold,
+            seq_threshold=seq_threshold,
+            nobias=kwargs["nobias"],
+            cpu=kwargs["cpu"],
+            binary=kwargs["hmmsearch"], 
+        )
+
+        # get rid of huge stdout log file immediately
+        try:
+            os.remove(ali.output)
+        except OSError:
+            pass
+
+        # turn namedtuple into dictionary to make
+        # restarting code nicer
+        ali = dict(ali._asdict())
+        # only item from hmmsearch_result to save is the hmmfile
+        ali["hmmfile"] = hmmfile
+
+        # save results of search for possible restart
+        write_config_file(ali_outcfg_file, ali)
+
```

Comment:
note to self: fixed all comments up to (not including) this on 1/3

Reply:
Actual reply: it would be easy to patch in the query sequence if I can go from .sto input -> fasta output. Otherwise I will have to get around to writing the .sto writer...